### PR TITLE
allow setting dict in command.identity , sweep.search_space in pipleine

### PR DESCRIPTION
**Pull request by lllilithyang**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 1]
```diff
@@ -240,6 +241,29 @@ def resources(self, value: Union[Dict, JobResourceConfiguration]):
             value = JobResourceConfiguration(**value)
         self._resources = value
 
+    @property
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:244]: please add some test for this change

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py [hunk 2]
```diff
@@ -149,6 +150,27 @@ def trial(self):
         """Id or instance of the command component/job to be run for the step."""
         return self._component
 
+    @property
+    def search_space(self):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py:154]: docstring

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 3]
```diff
@@ -240,6 +241,29 @@ def resources(self, value: Union[Dict, JobResourceConfiguration]):
...
+    def identity(
+        self,
+    ) -> Optional[Union[ManagedIdentityConfiguration, AmlTokenConfiguration, UserIdentityConfiguration]]:
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:247]: docstring

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 4]
```diff
@@ -240,6 +241,29 @@ def resources(self, value: Union[Dict, JobResourceConfiguration]):
...
+        return self._identity
+
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:249]: note: please resolve the pylint issues

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 5]
```diff
@@ -240,6 +241,29 @@ def resources(self, value: Union[Dict, JobResourceConfiguration]):
...
         self._resources = value
 
+    @property
+    def identity(
+        self,
+    ) -> Optional[Union[ManagedIdentityConfiguration, AmlTokenConfiguration, UserIdentityConfiguration]]:
+        return self._identity
+        
+    @identity.setter
+    def identity(self, value: Union[
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:251]: need to add some test for the new capability

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py (hunk 1 above)
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:244]: added in dsl/unittests/test_command_builder.py

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py (hunk 5 above)
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:251]: added in dsl/unittests/test_command_builder.py

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py (hunk 2 above)
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py:154]: added in search_space.setter

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py [hunk 6]
```diff
@@ -149,6 +150,30 @@ def trial(self):
...
+        return self._search_space
+
+    @search_space.setter
+    def search_space(self, values: Dict[str, Dict[str, Union[str, int, float, dict]]]):
+        """Support using dict to initialize search_space attribute value."""
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py:159]: No need to have docstring for setter, please do it for getter (which will be generated in reference doc site).

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py [hunk 7]
```diff
@@ -149,6 +150,30 @@ def trial(self):
         """Id or instance of the command component/job to be run for the step."""
         return self._component
 
+    @property
+    def search_space(self):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py:154]: docstring, copy what's in original component or job object : 

e.g.
Dictionary of the hyperparameter search space. The key is the name of the
        hyperparameter and the value is the parameter expression.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py (hunk 6 above)
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py:159]: deleted

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py [hunk 8]
```diff
@@ -240,6 +241,32 @@ def resources(self, value: Union[Dict, JobResourceConfiguration]):
...
+        return self._identity
+
+    @identity.setter
+    def identity(self, value: Union[
+                                Dict[str, str],
+                                ManagedIdentityConfiguration,
+                                AmlTokenConfiguration,
+                                UserIdentityConfiguration, None]):
+        if isinstance(value, dict):
+            identify_schema = UnionField(
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/command.py:260]: identify -> identity

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py [hunk 9]
```diff
@@ -320,3 +355,50 @@ def __setattr__(self, key, value):
...
+        if policy_dict is None:
+            self._early_termination = None
+        elif isinstance(policy_dict, EarlyTerminationPolicy):
+            self._early_termination = policy_dict
+        elif isinstance(policy_dict, dict):
+            assert 'type' in policy_dict.keys()
+            policy_type = policy_dict['type'].lower().capitalize()
+            if policy_type == EarlyTerminationPolicyType.BANDIT:
+                self._early_termination = BanditPolicy(
+                    delay_evaluation=policy_dict.get('delay_evaluation', 0),
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py:374]: why can't this be implemented as schema reserialize?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py [hunk 10]
```diff
@@ -320,3 +355,50 @@ def __setattr__(self, key, value):
...
+        return self._early_termination
+
+    @early_termination.setter
+    def early_termination(self, policy_dict: Union[EarlyTerminationPolicy, Dict[str, Union[str, float, int, bool]]]):
+        if policy_dict is None:
+            self._early_termination = None
+        elif isinstance(policy_dict, EarlyTerminationPolicy):
+            self._early_termination = policy_dict
+        elif isinstance(policy_dict, dict):
+            assert 'type' in policy_dict.keys()
```
[the remark this answers is not in the document]
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py]: I delete this logic. I find this logic could be replaced by UnionField([NestedField(schema)] and the related schemas  have beed defined.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py (hunk 9 above)
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/sweep.py]: modified

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**